sdk/storage/test/datalake/file_system_client_test.cpp

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "file_system_client_test.hpp"
+
+#include <algorithm>
+
+namespace Azure { namespace Storage { namespace Test {
+
+  const size_t c_PATH_TEST_SIZE = 5;
+
+  std::shared_ptr<DataLake::FileSystemClient> FileSystemClientTest::m_fileSystemClient;
+  std::string FileSystemClientTest::m_fileSystemName;
+
+  std::vector<std::string> FileSystemClientTest::m_pathNameSetA;
+  std::vector<std::string> FileSystemClientTest::m_pathNameSetB;
+  std::string FileSystemClientTest::m_directoryA;
+  std::string FileSystemClientTest::m_directoryB;
+
+  void FileSystemClientTest::SetUpTestSuite()
+  {
+    m_fileSystemName = LowercaseRandomString();
+    m_fileSystemClient = std::make_shared<DataLake::FileSystemClient>(
+        DataLake::FileSystemClient::CreateFromConnectionString(
+            ADLSGen2ConnectionString(), m_fileSystemName));
+    m_fileSystemClient->Create();
+
+    m_directoryA = LowercaseRandomString();
+    m_directoryB = LowercaseRandomString();
+    for (size_t i = 0; i < c_PATH_TEST_SIZE; ++i)
+    {
+      {
+        auto name = m_directoryA + LowercaseRandomString();
+        m_fileSystemClient->GetPathClient(m_directoryA + "/" + name).CreateFile();
+        m_pathNameSetA.emplace_back(std::move(name));
+      }
+      {
+        auto name = m_directoryB + LowercaseRandomString();
+        m_fileSystemClient->GetPathClient(m_directoryB + "/" + name).CreateFile();
+        m_pathNameSetB.emplace_back(std::move(name));
+      }
+    }
+  }
+
+  void FileSystemClientTest::TearDownTestSuite() { m_fileSystemClient->Delete(); }
+
+  std::vector<DataLake::Path> FileSystemClientTest::ListAllPaths(
+      bool recursive,
+      const std::string& directory)
+  {
+    std::vector<DataLake::Path> result;
+    std::string continuation;
+    DataLake::ListPathsOptions options;
+    if (!directory.empty())
+    {
+      options.Directory = directory;
+    }
+    do
+    {
+      auto response = m_fileSystemClient->ListPaths(recursive, options);
+      result.insert(result.end(), response.Paths.begin(), response.Paths.end());
+      if (response.Continuation.HasValue())
+      {
+        continuation = response.Continuation.GetValue();
+        options.Continuation = continuation;
+      }
+    } while (!continuation.empty());
+    return result;
+  }
+
+  DataLake::DataLakeHttpHeaders FileSystemClientTest::GetInterestingHttpHeaders()
+  {
+    static DataLake::DataLakeHttpHeaders result{
+        std::string("no-cache"),
+        std::string("attachment"),
+        std::string("deflate"),
+        std::string("en-US"),
+        std::string("application/octet-stream")};
```

Comment:
```suggestion
    static DataLake::DataLakeHttpHeaders result = [](){
        DataLake::DataLakeHttpHeaders ret;
        ret.Foo = "no-cache";
        ret.Bar = "attachment";
        return ret;
    }();

```

Reply:
Can you elaborate on why this resolves the issue? each call to result() will still create a map.

Reply:
`result` is a local variable, not a function. It cannot be **called**

Reply:
My bad. Did not see the last `()`.